addons/website_blog/static/src/js/options.js

```diff
@@ -107,7 +107,12 @@ options.registry.BlogPostTagSelection = options.Class.extend({
         if (!widgetValue) {
             return;
         }
-        const existing = Object.values(this.allTagsByID).some(tag => tag.name.toLowerCase() === widgetValue.toLowerCase());
+        const existing = Object.values(this.allTagsByID).some(
+            // A tag is already existing only if it was already defined (i.e.
+            // id is a number) or if it appears in the current list of tags.
+            tag => tag.name.toLowerCase() === widgetValue.toLowerCase() &&
+            (typeof(tag.id) === 'number' || this.tagIDs.includes(tag.id))
```

Comment:
Nitpicking: IMO the arrow on one line notation only is nice if on one line. In this case I would do:
```javascript
stuff.some(tag => {
    return tag_condition A
        && tag_condition B
        [...];
});